src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentQueue.cs

```diff
@@ -627,6 +627,18 @@ public bool TryPeek(out T result)
             return false;
         }
 
+        /// <summary>
+        /// Removes all objects from the <see cref="ConcurrentQueue{T}"/>.
+        /// </summary>
+        public void Clear()
+        {
+            while (!IsEmpty)
+            {
+                Segment head = _head;
+                head.Clear();
```

Comment:
TryDequeue remove only one element for call see `if (Interlocked.CompareExchange(ref _low, lowLocal + 1, lowLocal) == lowLocal)` code.

But head.Clear() call remove all elements in Segment see `if (Interlocked.CompareExchange(ref _low, highLocal + 1, lowLocal) == lowLocal)` code.